src/components/SearchBar.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, createElement } from "react";
+import { HybridConstraint, ListView, WrapperProps } from "./OfflineSearch";
+
+export interface SearchBarProps extends WrapperProps {
+    listView?: ListView;
+}
+
+interface SearchBarState {
+    buttonVisibility?: string;
+}
+
+export class SearchBar extends Component<SearchBarProps, SearchBarState> {
+    // internal variables
+    private searchButton: HTMLButtonElement;
+    private searchInput: HTMLInputElement;
+
+    constructor(props: SearchBarProps) {
+        super(props);
+
+        this.state = {
+            buttonVisibility: "hidden"
+        };
+        this.onClear = this.onClear.bind(this);
+        this.onSearchKeyDown = this.onSearchKeyDown.bind(this);
+    }
+
+    render() {
+        return this.props.showSearchBar
+            ?
+            createElement("div", { className: "search-container" },
```

Comment:
The root element is already specific as `widget-offline-search`, this is the search container inside the  `widget-offline-search` so it can stay as is